SU2_CFD/src/fluid/CCoolProp.cpp

```diff
@@ -56,28 +56,51 @@ void CCoolProp::SetTDState_rhoe(su2double rho, su2double e) {
   dTdrho_e = fluid_entity->first_partial_deriv(CoolProp::iT, CoolProp::iDmass, CoolProp::iUmass);
   dTde_rho = fluid_entity->first_partial_deriv(CoolProp::iT, CoolProp::iUmass, CoolProp::iDmass);
   if (fluid_entity->phase() == 6) {
+    // assume it is pure gas
+    fluid_entity->specify_phase(CoolProp::iphase_gas);
+    if (abs(Pressure / Pressure_Critical - 1) < 0.01) {
+      // if P is very cloase to Pc, the CoolProp has issue
+      Pressure = Pressure_Critical * 0.99;
+    }
+    fluid_entity->update(CoolProp::PT_INPUTS, Pressure, Temperature);
+    if (abs(fluid_entity->rhomass() / Density - 1) * 100 < 1) {
```

Comment:
It is good practice to define these thresholds (of 1%) as a constant of the class to allow tunning via a single parameter.

Reply:
Good idea.

I add a consant su2double **epsilon{0.01}** in _CoolProp.h_ to control the threshold for pressure.